SharedTools/Download-GitExtensions.ps1

```diff
@@ -0,0 +1,115 @@
+param([string] $Version, [string] $Source = "github")
+
+if (-not($Version)) 
+{ 
+    Throw "Parameter -Version is required";
+}
+
+if (-not($Source -eq "github") -and -not($Source -eq "appveyor"))
+{
+	Throw "Parameter -Source must be either 'github' or 'appveyor'";
+}
+
+Set-Location $PSScriptRoot
+
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+$ExtractRootPath = '../references'
+$AssetToDownloadName = 'GitExtensions-' + $Version + '.zip';
+$AssetToDownloadUrl = $null;
+
+$DownloadName = [System.IO.Path]::GetFileName($AssetToDownloadName);
+$DownloadFilePath = [System.IO.Path]::Combine($ExtractRootPath, $DownloadName);
+if (Test-Path $DownloadFilePath)
+{
+	Write-Host ('Download "' + $DownloadName + '" already exists.');
+	Pop-Location
+	exit;
+}
+
+Write-Host ('Searching for GitExtensions release on "' + $Source + '".');
+
+if ($Source -eq "github")
+{
+	# Find release and asset.
+	$Releases = Invoke-RestMethod -Uri 'https://api.github.com/repos/gitextensions/gitextensions/releases'
+	foreach ($Release in $Releases)
```

Comment:
This works, but this is not PowerShelly 😄  This can be written in a more functional way using pipes. 

Also you can iterate over the JSON in a more concise way. I'm sure you can use "." notation too.
Please have a look at 
* https://www.business.com/articles/using-powershell-with-json-data/
* https://devblogs.microsoft.com/scripting/playing-with-json-and-powershell/

I think the end result should look something like (pseudo code):
```
$AssetToDownloadUrl = $Releases | `
    Where-Object { $_.tag_name -eq $Version } | `
    Where-Object { $_.$assets.name -contains 'Portable' }
    Select-Object { $_.$assets.browser_download_url }
```

Reply:
Yes, I'm using PowerShell as C# 🙃

Reply:
You spend couple of weeks with me writing PS and you'll love it 🤓